ykrt/src/compile/mod.rs

```diff
@@ -0,0 +1,26 @@
+use crate::trace::IRTrace;
+use libc::c_void;
+use std::{error::Error, sync::Arc};
+use tempfile::NamedTempFile;
+
+#[cfg(jitc_llvm)]
+mod jitc_llvm;
+
+/// The trait that every JIT compiler backend must implement.
```

Comment:
I think we can avoid some of the naming problems by making this comment a bit more specific? What is a "JIT compiler backend"? Is it the backend that takes a mapped trace to an IR, or the backend that takes an IR to native code?

We do hope this distinction disappears in the end of course...

Apologies if I nitpick too much.